### ISSUE TITLE
Is it possible to add support for SQLITE in the cloud? per example turso.tech

### ISSUE THREAD

**Issue by MillhioreBT** [AuthMe/AuthMeReloaded#2873]
### What feature do you want to see added?

As the title says, would it be possible to add support for SQLite databases in the cloud? so we can use services with free tiers like [turso](https://turso.tech) or others that currently exist on the market.

### Are there any alternatives?

It is already explained in the previous text

### Anything else?

It is already explained in the previous text

Turso has an HTTPS API that can be used in this plugin to allow another method of sqlite database in the cloud.

![image](https://github.com/user-attachments/assets/f5672741-acd0-4374-8c53-e4d6a70f09b5)

Not only does Turso offer this, there are also other databases like Turso that offer this type of SQLite database in the cloud, in addition to having some advantages such as replicas in several countries.

**Comment by sgdc3** [comment]
@MillhioreBT what's the intended use case? SQLite is already supported as the default storage type and it is based on file storage.

**Comment by MillhioreBT** [comment]
> @MillhioreBT what's the intended use case? SQLite is already supported as the default storage type and it is based on file storage.

I know there is a method for SQLite, but not for SQLite for the cloud, for example if I want to access my SQLite database from my website?

There are also many companies that offer free SQLite database hosting in the cloud.

The advantage is that it is in the cloud and not in a local file.
This is very fashionable in the web world with current frameworks, and it would be nice to have a way to connect SQLite databases in the cloud with this great plugin.

For example, I have a Minecraft server that is basically made up of 3 servers, it is a network.
and the 3 servers should have access to this database, like my website, with a local sqlite file it is somewhat complicated and not practical for a network of servers.

It is perfect for a single server without a website, but for other cases SQLite is useless.

**Comment by games647** [comment]
>I know there is a method for SQLite, but not for SQLite for the cloud, for example if I want to access my SQLite database from my website?

Then you could use MySQL/MariaDB or PostgresSQL, etc. They are optimized for network use with cached statements, keep-alives, health checks, connection pooling (i.e. `HikariCP`), binary encoded messages instead of character encoded text from `HTTP Post`, type safety, etc. only to name a few. This feature would require us to use or develop a third-party connector that is not officially supported by the database platform.

Nevertheless, it should be mentioned that Firebase database (Cloud service for Apps from Google) does a similar thing with a NoSQL database. That is calling HTTP with JSON data as response. Although, this implementation uses a REST like endpoints and supports data listeners.

>The advantage is that it is in the cloud and not in a local file.
This is very fashionable in the web world with current frameworks, and it would be nice to have a way to connect SQLite databases in the cloud with this great plugin.

Many PHP and TS/JS support MySQL/MariaDB too.

In conclusion, implementing this connector would require a lot of work and maintenance to be worth included in this kind of project.

**Comment by MillhioreBT** [comment]
Thank you very much for the quick response.

I just thought it would be interesting and maybe catch someone's attention, but I understand that it's a lot of work and it's not worth it.
